benefits/core/models.py

```diff
@@ -251,6 +306,34 @@ def transit_processor_client_secret(self):
     def enrollment_flows(self):
         return self.enrollmentflow_set
 
+    def clean(self):
```

Comment:
Are we ok with this behavior:

- Create a new TransitAgency by providing a slug and leaving the rest as defaults
   - (agency is inactive)
- Create a new EnrollmentFlow by providing a system name and configure its `transit_agency` as the new TransitAgency, leave all the other fields as defaults
   - (flow has no configuration for claims or API verification)
- go back to the TransitAgency, fill out the required info fields, and make it active by checking the "Active" checkbox


Result: the agency is saved and made active, but the user might not realize the invalid EnrollmentFlow is connected to the active TransitAgency.

I'm not necessarily arguing for one way or the other, but just wanted clarity on if this is intentional.

Reply:
Ah, that is interesting... 

I have the `EnrollmentFlow.clean()` calling its `TransitAgency.clean()` -- but maybe it makes more sense the other way around?

Reply:
Oh I see. Yeah, since it's upon saving the TransitAgency to be active that we would want to make sure everything is valid, I think it makes more sense the other way around. 

I think this means we'd remove `EnrollmentFlow.clean()` calling its `TransitAgency.clean()`, and instead `TransitAgency.clean()` would call `EnrollmentFlow.clean()` for all its related `EnrollmentFlow`s.

Reply:
I'm curious what that validation would end up looking like in the UI. The user is looking at a `TransitAgency` form that does validation for some other objects :thinking: Hopefully it shows up in a reasonable way

Reply:
I'm curious too, I hadn't thought of a way to test this. But I think inverting the `clean()` gives us a way to test as you outline.

Reply:
Hmm well, not good news

I inverted the `clean()` and now we can trigger the condition, but, it's an error in Django Admin since the fields from `EnrollmentFlow` don't exist on `TransitAgency`:

![image](https://github.com/user-attachments/assets/3eb49d81-ae4c-4fe0-b6ab-336d0d45f841)

So I think we have to remove the `clean()` entirely.

Reply:
Hmm I see. Seems like what's happening is `EnrollmentFlow.clean()` raises field-level `ValidationError`s which then `TransitAgencyForm` doesn't have fields for, so understandably the `ValueError` is raised.

I started thinking if perhaps we could catch the error from `EnrollmentFlow.clean()` and have logic to handle it in `TransitAgency.clean()`, so something like this:
```python
        for flow in self.enrollment_flows.all():
            try:
                flow.clean()
            except Exception:
                raise ValidationError("There was an error with a related EnrollmentFlow.")
```

We could do more to make the message more helpful to the user (like somehow including the `EnrollmentFlow` system name), but to avoid scope creep, this simple message would at least help the user not activate a TransitAgency with invalid EnrollmentFlows.

Here's what it looks like when you try to activate the `TransitAgency`:
![Screenshot from 2024-11-26 17-05-18](https://github.com/user-attachments/assets/aad32d75-5fca-4a76-aae2-080105df2335)

Reloading the page shows that the agency was not activated. (I reloaded the page by selecting the URL bar in my browser and hitting Enter to make sure it's a GET request)

![Screenshot from 2024-11-26 17-14-36](https://github.com/user-attachments/assets/3b51f092-1316-4432-9500-7a7d61056bba)

Reply:
[`a8c40bb`](https://github.com/cal-itp/benefits/pull/2471/commits/a8c40bb6e9f95146790f6bccf7d871f3037d2190)